execution_engine/src/core/engine_state/mod.rs

```
@@ -442,9 +445,40 @@ where
             tracking_copy.borrow_mut().force_write(*key, value.clone());
         }
 
+        // Perform global state migrations that require state.
+
+        if let Some(activation_point) = upgrade_config.activation_point() {
+            // Activation point is set to the next era after the current one.
+            let highest_era_info_id = activation_point.saturating_sub(1);
+
+            // Start of the copy
+            let highest_era_info_key = Key::EraInfo(highest_era_info_id);
+
+            let get_result = tracking_copy
+                .borrow_mut()
+                .get(correlation_id, &highest_era_info_key)
+                .map_err(|error| Error::Exec(error.into()))?;
+
+            match get_result {
+                Some(stored_value @ StoredValue::EraInfo(_)) => {
+                    tracking_copy
+                        .borrow_mut()
+                        .force_write(Key::EraSummary, stored_value);
+                }
+                Some(_other_stored_value) => {
+                    // This should not happen as we only write EraInfo variants.
```

Shouldn't we return an error here?  Or at least log an error?

---

Added an error variant in `ProtocolUpgradeError` and logged error with `error!`